server/src/main/java/org/elasticsearch/index/shard/PrimaryReplicaSyncer.java

```diff
@@ -83,6 +84,8 @@ public void resync(final IndexShard indexShard, final ActionListener<ResyncTask>
         ActionListener<ResyncTask> resyncListener = null;
         try {
             final long startingSeqNo = indexShard.getGlobalCheckpoint() + 1;
+            final SeqNoStats seqNoStats = indexShard.seqNoStats();
+            final long maxSeqNo = seqNoStats != null ? seqNoStats.getMaxSeqNo() : SequenceNumbers.UNASSIGNED_SEQ_NO;
```

Comment:
otherwise PrimaryReplicaSyncerTests#testSyncerOnClosingShard fails with NPE there, as I understood it happens on primary goes down and tries to send syncrequest out.
 `./gradlew null -Dtests.seed=89AD08B811599795 -Dtests.class=org.elasticsearch.index.shard.PrimaryReplicaSyncerTests -Dtests.method="testSyncerOnClosingShard"`

Reply:
ok, looking into that